lib/features/steps/multipart_request_steps.rb

```diff
@@ -0,0 +1,126 @@
+  parts = Server.current_request[:body]
+  assert_equal(part_count, parts.size)
+end
+
+# Tests the multipart request has at least one field.
+Then('the multipart request has a non-empty body') do
+  parts = Server.current_request[:body]
+  assert(parts.size.positive?, "Multipart request payload contained #{parts.size} fields")
+end
+
+# Tests that a multipart request field exists and is not null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the multipart field {string} is not null') do |part_key|
+  parts = Server.current_request[:body]
+  assert_not_nil(parts[part_key], "The field '#{part_key}' should not be null")
+end
+
+# (Deprecated) Retained for backwards compatibility
+# Tests that a multipart request field exists and is not null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the field {string} for multipart request is not null') do |part_key|
+  step %(the multipart field "#{part_key}" is not null)
+end
+
+# Tests that a multipart request field equals a string.
+#
+# @step_input part_key [String] The key to the multipart element
+# @step_input expected_value [String] The string to match against
+Then('the multipart field {string} equals {string}') do |part_key, expected_value|
+  parts = Server.current_request[:body]
+  assert_equal(parts[part_key], expected_value)
+end
+
+# (Deprecated) Retained for backwards compatibility
+# Tests that a multipart request field equals a string.
+#
+# @step_input part_key [String] The key to the multipart element
+# @step_input expected_value [String] The string to match against
+Then('the field {string} for multipart request equals {string}') do |part_key, expected_value|
+  step %(the multipart field "#{part_key}" equals "#{expected_value}")
+end
+
+# Tests that a multipart request field is null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the multipart field {string} is null') do |part_key|
+  parts = Server.current_request[:body]
+  assert_nil(parts[part_key], "The field '#{part_key}' should be null")
+end
+
+# (Deprecated) Retained for backwards compatibility
+# Tests that a multipart request field is null.
+#
+# @step_input part_key [String] The key to the multipart element
+Then('the field {string} for multipart request is null') do |part_key|
+  step %(the multipart field "#{part_key}" is null)
+end
+
+# Tests that the multipart payload body does not match a JSON fixture.
+# JSON formatted multipart fields will be parsed into hashes.
+#
+# @step_input fixture_path [String] Path to a JSON fixture
+Then('the multipart body does not match the JSON fixture in {string}') do |fixture_path|
+  raw_payload_value = Server.current_request[:body]
+  payload_value = {}
+  raw_payload_value.each_with_object(payload_value) do |(k, v), out|
+    out[k] = JSON.parse(v.to_s)
+  rescue JSON::ParserError
+    out[k] = v.to_s
+  end
+  expected_value = JSON.parse(open(fixture_path, &:read))
```

Comment:
I think (my memory is terrible, sorry!) part of the reason I wrote new steps was because I couldn't figure out where this was relative to, so I made mine use an absolute path instead — it'd be good to have docs for where the JSON file should be/what the CWD is (and possible be able to change that)

IMO it's also nice for these files not to be in `features/fixtures`, because some repos will have a lot of different fixtures, so it's easier to see when they are separate (I put them in `features/expected`)

Reply:
From some quick testing the `fixture_path` doesn't need to be in `features/fixtures`.  I'll change the variable name and doc string to resolve the issue.